src/Coordination/KeeperDispatcher.cpp

```diff
@@ -194,6 +184,28 @@ void KeeperDispatcher::requestThread()
                     prev_result = result;
                 }
 
+                /// If we will execute read or reconfig next, we have to process result now
+                if (has_read_request || has_reconfig_request)
+                {
+                    if (prev_result)
+                        result_buf = forceWaitAndProcessResult(prev_result, current_batch);
+
+                    if (result_buf)
+                    {
+                        nuraft::buffer_serializer bs(result_buf);
+                        auto log_idx = bs.get_u64();
+
+                        while (true)
```

Comment:
We have this loop because leader may have already apply `log_idx` but we didn't do it yet?